### PR TITLE
Use Storage Access Framework for backup/restore and update german translation

### LINKED ISSUE #26: Text field for preferences backup naming

### PR DESCRIPTION
**Pull request by FliegendeWurst**
This fixes #26 on newer Android versions.

### CODE REVIEW

File: app/src/main/java/ru/henridellal/emerald/BackupPreference.java [hunk 1]
```diff
@@ -19,33 +16,44 @@ public BackupPreference(Context c) {
...
+
 	@Override
 	protected View onCreateDialogView() {
 		int mode = 0;
 		if ("backup".equals(getKey())) {
-				mode = 0;
+			mode = 0;
 		} else if ("restore".equals(getKey())) {
-				mode = 1;
-		} else if ("convert".equals(getKey())) {
```
Comment (FliegendeWurst) [inline app/src/main/java/ru/henridellal/emerald/BackupPreference.java]: I removed this because I couldn't find any usage of this key

Comment (HenriDellal) [reply app/src/main/java/ru/henridellal/emerald/BackupPreference.java]: I've used this mode to check the database content. It isn't needed anymore.

**Comment by HenriDellal** [comment]
Thank you very much. I'll test it and merge then. Please could you tell if it is tested or not?

**Comment by FliegendeWurst** [comment]
I tested the new method and the old one on my Huawei p8 lite (Android 6). They both work.

**Pull request merged**